src/Microsoft.Azure.WebJobs.Host.Storage/HostStorageProvider.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Azure.Storage.Blobs;
+using Microsoft.Azure.WebJobs.StorageProvider.Blobs;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs
+{
+    public class HostStorageProvider
```

Comment:
Can we add Try** methods to the `BlobServiceClientProvider` and avoid an extra level of the hierarchy?